docs/current/sdk/go/snippets/work-with-host-filesystem/export-dir/main.go

```diff
@@ -0,0 +1,42 @@
+package main
+
+import (
+  "context"
+  "log"
+  "fmt"
+  "os"
+  "path/filepath"
+
+  "dagger.io/dagger"
+)
+
+func main() {
+  hostdir := os.TempDir()
+
+  ctx := context.Background()
+
+  client, err := dagger.Connect(ctx)
+  if err != nil {
+    log.Fatal(err)
+  }
+  defer client.Close()
+
+  // highlight-start
+  containerdir, err := client.Container().From("alpine:latest").Directory("/tmp").
+    WithNewFile("foo.txt", dagger.DirectoryWithNewFileOpts{
+      Contents: "this is foo",
+    }).ID(ctx)
+  if err != nil {
+    log.Fatal(err)
+  }
+
+  client.Directory(dagger.DirectoryOpts{ID: containerdir}).Export(ctx, hostdir)
+  // highlight-end
+
+  contents, err := os.ReadFile(filepath.Join(hostdir, "foo.txt"))
+  if err != nil {
+    log.Fatal(err)
+  }
```

Comment:
This example felt a bit misleading, since it doesn't actually create an entry under `/tmp` in the container, it just took the container's directory and created a new directory with content added to it.

The distinction here is very subtle and I think worth avoiding for the first example. Essentially by adding a file to the returned directory, the original container is unmodified; you just have a new directory that would have to be added back to the container.

Here's a replacement that modifies the container directly:

```suggestion
  _, err := client.Container().From("alpine:latest").
    WithWorkdir("/tmp").
    Exec(dagger.ContainerExecOpts{
      Args: []string{"wget", "https://dagger.io"},
    }).
    Directory(".").
    Export(ctx, hostdir)
  if err != nil {
    log.Fatal(err)
  }
  // highlight-end
  contents, err := os.ReadFile(filepath.Join(hostdir, "index.html"))
  if err != nil {
    log.Fatal(err)
  }
  fmt.Println(string(contents))
```